src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -946,6 +946,54 @@ mod tests {
         assert!(pool.get_filesystem_by_uuid(fs_uuid).is_none());
     }
 
+    #[test]
+    pub fn loop_test_meta_expand() {
+        // This test requires more than 1 GiB.
+        loopbacked::test_with_spec(loopbacked::DeviceLimits::Range(2, 3), test_meta_expand);
+    }
+
+    #[test]
+    pub fn real_test_meta_expand() {
+        real::test_with_spec(real::DeviceLimits::Range(2, 3), test_meta_expand);
+    }
+
+    fn test_meta_expand(paths: &[&Path]) -> () {
+        let pool_uuid = Uuid::new_v4();
+        let dm = DM::new().unwrap();
+        let mut mgr = BlockDevMgr::initialize(pool_uuid, paths, MIN_MDA_SECTORS, false).unwrap();
+        // Create a ThinPool with a very small meta device.
+        let mut thin_pool = ThinPool::new(pool_uuid,
+                                          &dm,
+                                          &ThinPoolSizeParams {
```

Comment:
Here you can use strange Default syntax:
```
&ThinPoolSizeParams {
    meta_size: MetaBlocks(16),
    ...Default::default(),
}
```

Reply:
Got this syntax from https://stackoverflow.com/questions/19650265/is-there-a-faster-shorter-way-to-initialize-variables-in-a-rust-struct, but I think I originally ran across it in programming Rust book.

Reply:
ok